server/routes/api/users.js

```diff
@@ -3,20 +3,51 @@ const express = require("express");
 const jwt = require("jsonwebtoken");
 const bcrypt = require("bcrypt");
 const configureStripe = require("stripe");
+const multer = require("multer");
+const multers3 = require("multer-s3");
+const AWS = require("aws-sdk");
 
-const { passportSecret, stripeSecretKey } = require("../../config/keys");
+const {
+  passportSecret,
+  stripeSecretKey,
+  awsBucketName,
+  awsAccessKeyId,
+  awsSecretAccessKey,
+  awsRegion,
+} = require("../../config/keys");
 const User = require("../../models/User");
 const validateEmail = require("../../validation/email");
 const validatePassword = require("../../validation/password");
 const validateExperience = require("../../validation/experience");
 const authenticate = require("../../middlewares/authenticate");
+const { newImage } = require("../../controllers/profileImage");
 
 const stripe = configureStripe(stripeSecretKey);
 
 const router = express.Router();
 
 const price = 100; // the USD price (in cents) of 1 credit
 
+const s3 = new AWS.S3({
+  accessKeyId: awsAccessKeyId,
+  secretAccessKey: awsSecretAccessKey,
+  region: awsRegion,
+});
+
+const uploadS3 = multer({
+  storage: multers3({
+    s3: s3,
+    acl: "public-read",
+    bucket: awsBucketName,
+    metadata: (req, file, cb) => {
+      cb(null, { fieldName: file.fieldname });
+    },
+    key: (req, file, cb) => {
+      cb(null, Date.now().toString() + "-" + file.originalname);
+    },
+  }),
+});
+
```

Comment:
Could probably move this into its own file, maybe in the services directory, that exports the uploadS3 object.